learning/tour-of-beam/frontend/lib/pages/tour/widgets/unit_content.dart

```diff
@@ -30,6 +32,76 @@ class UnitContentWidget extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return MarkdownBody(data: unitContent.description);
+    final textTheme = Theme.of(context).textTheme;
+    final markdownStyle = MarkdownStyleSheet(
+      h1: textTheme.headlineLarge,
+      h3: textTheme.headlineMedium,
+      p: textTheme.bodyMedium,
+      code: Theme.of(context)
+          .extension<BeamThemeExtension>()!
+          .codeRootStyle
+          .copyWith(
+            backgroundColor: BeamColors.transparent,
+            fontSize: BeamSizes.size12,
+          ),
+      codeblockDecoration: BoxDecoration(
+        // TODO(nausharipov): where to store these colors?
+        color: Theme.of(context).brightness == Brightness.dark
+            ? BeamDarkThemeColors.codeBackground
+            : BeamLightThemeColors.codeBackground,
+        border: Border.all(color: Theme.of(context).primaryColor),
+        borderRadius: const BorderRadius.all(
+          Radius.circular(BeamSizes.size4),
+        ),
+      ),
+    );
+
+    return Markdown(
+      selectable: true,
+      data: unitContent.description,
+      builders: {
+        'code': CodeBuilder(),
+      },
+      styleSheet: markdownStyle,
+    );
+  }
+}
+
+class CodeBuilder extends MarkdownElementBuilder {
+  @override
+  Widget? visitElementAfter(md.Element element, TextStyle? preferredStyle) {
+    final String textContent = element.textContent;
+    final bool isMultilineCode = textContent.contains('\n');
```

Comment:
Can we check it the element is in single backticks and triple backticks instead of looking for `\n`? There may be single lines formatted as codeblocks. I guess they will still contain `\n` at the end, but the number of ticks is an immediate indicator and so it feels more reliable.

Reply:
I couldn't find a way to check for the number of backticks in `MarkdownElementBuilder`. 

<img width="864" alt="Screen Shot 2022-10-18 at 13 31 51" src="https://user-images.githubusercontent.com/31556582/196365565-90f1d44e-08c0-467b-94f9-e35e1d419262.png">

Reply:
I've changed '\n' to four spaces to distinguish code blocks, relying on `markdown` documentation:

<img width="998" alt="Screen Shot 2022-10-18 at 14 00 40" src="https://user-images.githubusercontent.com/31556582/196372415-af3e1f47-d61c-450d-84a1-97ebc58f4273.png">

Reply:
This comment is irrelevant to our task.
Please revert to `\n` detection, it is more reliable.